### ISSUE TITLE
File Integrity Check takes ages because of (non-default) visible .zfs snapdir

### ISSUE THREAD

**Issue by frijsdijk** [owncloud/core#26400]
### Steps to reproduce

Prerequisite: Owncloud runs on it's own zfs mountpoint (zfs create -o mountpoint=/var/www/owncloud tank/owncloud) - so there is a /var/www/owncloud/.zfs directory where ZFS's snapshots can be found. This directory is hidden by default (but you can 'cd' into it).

1. make ZFS snapdir visible (zfs set snapdir=visible tank/owncloud
2. trigger file integrity check
3. Owncloud's file integrity check will also go into /var/www/owncloud/.zfs/snapshot/*/*, and from there on also starts reading /var/www/owncloud/.zfs/snapshot/*/data/, which, depending on how much data and snapshots there are, can take ages.

### Expected behaviour
Owncloud's integrity checker should ignore/not go into .zfs directories. This directory has special properties (it's readonly, touching anything in there will give a 'operation not supported' error in FreeBSD).

### Actual behaviour
Owncloud's integrity checker started reading the .zfs dir causing the process to consume alot of disk io resources and never seemed to end (I have 4 week, twice a day snapshots in there, and about 66G data).

### Server configuration
**Operating system**: FreeBSD 10.3-RELEASE-p4
**Web server:** Apache/2.4.23
**Database:** percona56-server-5.6.32.78.0
**PHP version:** php70-7.0.11 (FPM)
**ownCloud version:** (see ownCloud admin page) 9.1.0 -> 9.1.1
**Updated from an older ownCloud or fresh install:** Fresh install
**Where did you install ownCloud from:** FreeBSD ports
**Signing status (ownCloud 9.0 and above):** ?

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.

None now. After hiding the .zfs dir, the integrity checker ran correctly (before that I didn't get results as it took very long (started debugging after 10 or so minutes).

```

**List of activated apps:** 

# su -m www -c 'php occ app:list'

Enabled:
  - activity: 2.3.2
  - comments: 0.3.0
  - dav: 0.2.6
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - notifications: 0.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.1
  - updatenotification: 0.2.1
Disabled:
  - encryption
  - external
  - files_antivirus
  - files_external
  - user_external
  - user_ldap

**The content of config/config.php:**

{
    "system": {
        "instanceid": "ocu1yl842427",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "owncloud.isafeelin.org"
        ],
        "datadirectory": "\/usr\/local\/www\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/owncloud.isafeelin.org",
        "dbtype": "mysql",
        "version": "9.1.1.3",
        "dbname": "oc_central",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "theme": "",
        "loglevel": 0,
        "maintenance": false
    },
    "apps": {
        "activity": {
            "enabled": "yes",
            "installed_version": "2.3.2",
            "types": "filesystem"
        },
        "backgroundjob": {
            "lastjob": "161"
        },
        "comments": {
            "enabled": "yes",
            "installed_version": "0.3.0",
            "types": "logging"
        },
        "core": {
            "installedat": "1469216492.6424",
            "lastcron": "1476811649",
            "lastupdateResult": "[]",
            "lastupdatedat": "1476810686",
            "oc.integritycheck.checker": "[]",
            "public_files": "files_sharing\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "repairlegacystoragesdone": "yes"
        },
        "dav": {
            "enabled": "yes",
            "installed_version": "0.2.6",
            "types": "filesystem"
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "0.3.0",
            "types": ""
        },
        "federation": {
            "enabled": "yes",
            "installed_version": "0.1.0",
            "types": "authentication"
        },
        "files": {
            "cronjob_scan_files": "500",
            "enabled": "yes",
            "installed_version": "1.5.1",
            "types": "filesystem"
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "0.8.1",
            "ocsid": "166049",
            "types": ""
        },
        "files_sharing": {
            "enabled": "yes",
            "installed_version": "0.10.0",
            "types": "filesystem"
        },
        "files_texteditor": {
            "enabled": "yes",
            "installed_version": "2.1",
            "ocsid": "166051",
            "types": ""
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "0.9.0",
            "types": "filesystem"
        },
        "files_versions": {
            "enabled": "yes",
            "installed_version": "1.3.0",
            "types": "filesystem"
        },
        "files_videoplayer": {
            "enabled": "yes",
            "installed_version": "0.9.8",
            "types": ""
        },
        "firstrunwizard": {
            "enabled": "yes",
            "installed_version": "1.1",
            "ocsid": "166055",
            "types": ""
        },
        "gallery": {
            "enabled": "yes",
            "installed_version": "15.0.0",
            "types": ""
        },
        "notifications": {
            "enabled": "yes",
            "installed_version": "0.3.0",
            "types": "logging"
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "0.5.0",
            "types": "prevent_group_restriction"
        },
        "systemtags": {
            "enabled": "yes",
            "installed_version": "0.3.0",
            "types": "logging"
        },
        "templateeditor": {
            "enabled": "yes",
            "installed_version": "0.1",
            "types": ""
        },
        "updatenotification": {
            "enabled": "yes",
            "installed_version": "0.2.1",
            "types": ""
        }
    }
}

**Are you using external storage, if yes which one:** local/smb/sftp/...  NO

**Are you using encryption:** yes/no NO

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/... NO

### Client configuration
**Browser:** Mozilla/5.0 (Windows NT 10.0; WOW64; rv:49.0) Gecko/20100101 Firefox/49.0

**Operating system:** Win10

**Comment by PVince81** [comment]
Closing in favor of this ticket https://github.com/owncloud/core/issues/24218 which is more generic, about providing a way to exclude known patterns from the check.

**Comment by PVince81** [comment]
Actually I rather mean this one https://github.com/owncloud/core/issues/23186 which is about making it possible to add exceptions for integrity check in a config file.
